contracts/RouterStrategy.sol

```diff
@@ -0,0 +1,232 @@
+// SPDX-License-Identifier: AGPL-3.0
+
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import {BaseStrategy} from "@yearnvaults/contracts/BaseStrategy.sol";
+import {
+    SafeERC20,
+    SafeMath,
+    IERC20,
+    Address
+} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+
+interface IVault is IERC20 {
+    function token() external view returns (address);
+
+    function decimals() external view returns (uint256);
+
+    function deposit() external;
+
+    function pricePerShare() external view returns (uint256);
+
+    function withdraw(
+        uint256 amount,
+        address account,
+        uint256 maxLoss
+    ) external returns (uint256);
+}
+
+contract RouterStrategy is BaseStrategy {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    string internal strategyName;
+    IVault public yVault;
+    uint256 public maxLoss;
+
+    constructor(
+        address _vault,
+        address _yVault,
+        string memory _strategyName
+    ) public BaseStrategy(_vault) {
+        _initializeThis(_yVault, _strategyName);
+    }
+
+    function _initializeThis(address _yVault, string memory _strategyName)
+        internal
+    {
+        yVault = IVault(_yVault);
+        strategyName = _strategyName;
+    }
+
+    function name() external view override returns (string memory) {
+        return strategyName;
+    }
+
+    function estimatedTotalAssets() public view override returns (uint256) {
+        return balanceOfWant().add(valueOfInvestment());
+    }
+
+    function delegatedAssets() external view override returns (uint256) {
+        return vault.strategies(address(this)).totalDebt;
+    }
+
+    function prepareReturn(uint256 _debtOutstanding)
+        internal
+        override
+        returns (
+            uint256 _profit,
+            uint256 _loss,
+            uint256 _debtPayment
+        )
+    {
+        uint256 _totalDebt = vault.strategies(address(this)).totalDebt;
+        uint256 _totalAsset = estimatedTotalAssets();
+
+        // Estimate the profit we have so far
+        if (_totalDebt <= _totalAsset) {
+            _profit = _totalAsset.sub(_totalDebt);
+        }
```

Comment:
can you give an example? it would be considered in `liquidatePosition()`

Reply:
Consider when 042 vault suffers a loss/hack. 

Pps would still show as healthy on old vault A, but a withdraw would suffer a loss when pulling from new vault.

Could block withdraws from web page if maxLoss is exceeded.

Reply:
very good point @retro-wavey 

thinking about this, some thoughts: 
1. if we consider vaults as up-only, having a loss should be considered as a special case
2. when a hack happens (rare thing), either we prefer to hold people from withdrawing (although is still posible if they take the losses, setting maxLoss higher) or we will reimburse, making pps to go up and erasing the issue after applying a solution. In any case, this "bug" would buy us time
3. We can set maxLoss to 10_000 from this vault to avoid being stopped from withdrawing from newer vault
4. Adding the else clause would mean that if we harvest we take the loss no matter what happens afterward

im not sure what would I do here but for sure I have pros and cons for both approaches

IMO it is cleaner if we add the else clause but as it is it would work for sure. 

taking into account that in any case we need to harvest (either realising the loss through a lower debt ratio) or harvesting to report the loss, it may be simpler/safer to not add the else clause (as we would report losses 100% on purpose)

Reply:
Good comments.

Personally I am fine either way. 

Poolpi, you have my LGTM.

Reply:
cool, this is great. will mark as resolved and merge.